category/vm/compiler/ir/x86/emitter.cpp

```diff
@@ -5797,6 +5823,14 @@ namespace monad::vm::compiler::native
         }
         // Case 8: (stack, stack)
         if (dst->stack_offset() && src->stack_offset()) {
+            if (dst->stack_offset().value().moved_from == PrevLoc::GprReg) {
```

Comment:
Also prefer `GeneralReg` when `src` was previously in GPR
```suggestion
            if (dst->stack_offset()->moved_from == PrevLoc::GprReg ||
                 src->stack_offset()->moved_from == PrevLoc::GprReg) {
```